lib/internal/Magento/Framework/View/Layout.php

```diff
@@ -545,8 +545,7 @@ public function renderNonCachedElement($name)
             if ($this->appState->getMode() === AppState::MODE_DEVELOPER) {
                 throw $e;
             }
-            $message = ($e instanceof LocalizedException) ? $e->getLogMessage() : $e->getMessage();
-            $this->logger->critical($message);
+            $this->logger->critical($e);
```

Comment:
Basically, I like this approach much better than the existing one. However, we will miss one piece of the flow with this change. When we work with `getLogMessage()` method of `LocalizedException` the error handler fills out the placeholders with values in case if we have something like 

```
Cannot render %1
```

If we simply pass the exception to the logger, the default `__toString` implementation of the `\Exception` will not do the same trick.

Reply:
That's a valid point. If that is the case we can either do something like this: 
```
            $message = ($e instanceof LocalizedException) ? $e->getLogMessage() : $e->getMessage();
            $this->logger->critical($message, ['exception' => $e]);
```
To make the exception stack log.
Or even better I can change `LoggerInterface::critical` implementation in this PR so that it will use `getLogMessage` if type is `LocalizedException`. Does that make sense?